### DISCUSSION TITLE
shopDomainUpdate only via graphql?

### DISCUSSION THREAD

**Discussion by genox** [saleor/saleor#13968] [Q&A] [answered]
Hi,

as part of self-hosted setups we are supposed to change the domain in the configuration of the api instance. The docs refer to the following graphql query:

```graphql
mutation UpdateDomain {
  shopDomainUpdate(input: { domain: "api.example.com" }) {
    shop {
      domain {
        host
      }
    }
  }
}
```

Is there no way to run this command somehow via CLI on the docker container itself? Having to get an admin auth token first from a browser's debug tools and then either clobbering together a curl POST or using the graphql web client seems a little bit complicated for a simple configuration change that is a text input on a settings page on 99% of all the other web apps. :-)

The setup right now is mainly based on running commands directly in python scripts, manage.py, which can be run with docker run / exec. The step of changing the shop domain is also missing on the docker images page of the documentation.

**Comment by timuric** [comment, marked as answer]
@genox generally we don't think shop domain belongs in Saleor configuration, in headless setup there can be many domains and clients talking to Saleor. I recommend storing this configuration elsewhere, for example global config for your integrations and apps.
